automatic/adobereader/tools/chocolateyInstall.ps1

```diff
@@ -1,90 +1,94 @@
-  4106="es_ES"; # 4106 = Spanish - Guatemala
-  18442="es_ES"; # 18442 = Spanish - Honduras
-  22538="es_ES"; # 22538 = Spanish - Latin America
-  2058="es_ES"; # 2058 = Spanish - Mexico
-  19466="es_ES"; # 19466 = Spanish - Nicaragua
-  6154="es_ES"; # 6154 = Spanish - Panama
-  15370="es_ES"; # 15370 = Spanish - Paraguay
-  10250="es_ES"; # 10250 = Spanish - Peru
-  20490="es_ES"; # 20490 = Spanish - Puerto Rico
-  21514="es_ES"; # 21514 = Spanish - United States
-  14346="es_ES"; # 14346 = Spanish - Uruguay
-  8202="es_ES"; # 8202 = Spanish - Venezuela
-}
-$LCID = (Get-Culture).LCID
-if ($mapping.ContainsKey($LCID)) {
-  $locale = $mapping.Get_Item($LCID);
-  Write-Host "Switching to $locale"
-  $url = $url -replace 'en_US', "$locale"
+﻿try {
+
+$arguments = @{}
+
+  # Now we can use the $env:chocolateyPackageParameters inside the Chocolatey package
+  $packageParameters = $env:chocolateyPackageParameters
+
+  # Default value
+  $lcid = $null
+
+
+  # Now parse the packageParameters using good old regular expression
+  if ($packageParameters) {
+      $match_pattern = "\/(?<option>([a-zA-Z0-9]+)):(?<value>([`"'])?([a-zA-Z0-9- \(\)\s_\\:\.]+)([`"'])?)|\/(?<option>([a-zA-Z]+))"
+      $option_name = 'option'
+      $value_name = 'value'
+
+      if ($packageParameters -match $match_pattern ){
+          $results = $packageParameters | Select-String $match_pattern -AllMatches
+          $results.matches | % {
+            $arguments.Add(
+                $_.Groups[$option_name].Value.Trim(),
+                $_.Groups[$value_name].Value.Trim())
+        }
+      }
+      else
+      {
+          Throw "Package Parameters were found but were invalid (REGEX Failure)"
+      }
+
+      if ($arguments.ContainsKey("lcid")) {
+          Write-Host "lcid Argument Found"
+          $lcid = $arguments["lcid"]
+          echo "LCID set to $lcid"          
+      } 
+
+
+  } else {
+      Write-Debug "No Package Parameters Passed in"
+  }
+
+  $scriptDir = $(Split-Path -parent $MyInvocation.MyCommand.Definition)
+  Import-Module (Join-Path $scriptDir 'checkinstall.ps1')
+  #If no Language ID passed through, use the one from Windows
+  #LCID table can be found at http://msdn.microsoft.com/es-es/goglobal/bb964664.aspx to manually pass through the desired language. 
+   if (!$lcid){ 
+    $lcid = (Get-Culture).LCID
+    echo "LCID set to $lcid"  
+   }
+
+  
+  # Find download URLs at http://www.adobe.com/support/downloads/product.jsp?platform=windows&product=10
+  
+  $url = 'http://ardownload.adobe.com/pub/adobe/reader/win/AcrobatDC/1500720033/AcroRdrDC1500720033_MUI.exe'
+  $version = "15.007.20033"
+
+
+  $packageName = 'adobereader'
+  $installerType = 'exe'
+  $installArgs = "/sALL /rs /sl $lcid"
+  
+ #Kill the Adobe Updater AdobeARM.exe
+  
+  Stop-Process -processname AdobeARM -ErrorAction SilentlyContinue
+
+  #Check for current version installed
+
+  #Check for OS bitness
+if ((Get-WmiObject Win32_OperatingSystem).OSArchitecture -match "64-bit"){

```

Comment:
We have something that already does this `Get-ProcessorBits` - horribly named but it's about the OS bitness. https://github.com/chocolatey/choco/wiki/HelpersReference#complete-list-alphabetical-order